swagger_server/test/e2e/test_samples_get_ids_controller.py

```diff
@@ -0,0 +1,61 @@
+from hypothesis import given, strategies as st, reproduce_failure
+
+from swagger_server.test.strategies import samples, experiment_ids
+
+
+def test_samples_get_without_any_id(get_sample_by_ids):
+    assert get_sample_by_ids('').status_code == 404
+
+
+@given(sample_ids=st.lists(min_size=1, max_size=5, elements=experiment_ids()))
+def test_samples_get_non_existent_ids(sample_ids, get_sample_by_ids):
+    assert get_sample_by_ids(",".join(sample_ids)).status_code == 404
+
+
+@given(samples_with_ids=st.lists(min_size=1, max_size=5, elements=samples(), unique_by=lambda x: x.experiment_id))
+def test_samples_get_existing_ids(samples_with_ids, create_sample, create_leaf, get_sample_by_ids, sample_graph):
+    try:
+        created_ids = []
+        for sample in samples_with_ids:
+            if sample.nearest_neighbours:
+                for neighbour in sample.nearest_neighbours:
+                    create_sample(neighbour)
+            if sample.nearest_leaf_node:
+                create_leaf(sample.nearest_leaf_node)
+            create_sample(sample)
+            created_ids.append(sample.experiment_id)
+
+        response = get_sample_by_ids(",".join(created_ids), ensure=True)
```

Comment:
This is not the usage of `ensure=True`. `ensure` is a way to express that we implicitly expect these calls to be successful and we don't need to assert their status codes after calling.

Therefore, in a test, we want to put `ensure=True` in the fixture calls that are supposed to setup the test, and only explicitly assert the status of the call we want to test (since we want that to be explicit)

Reply:
Ok, I have status code checking after this call, so this is redundant. I will remove it.

Reply:
Fixed in 74fe794